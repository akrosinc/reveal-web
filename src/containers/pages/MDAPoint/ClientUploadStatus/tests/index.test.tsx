import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import flushPromises from 'flush-promises';
import React from 'react';
import UploadStatus from '..';

describe('components/ClientUploadStatus', () => {
  const file = new File(['student'], 'student.csv', {
    type: 'text/csv',
  });
  it('renders without crashing', () => {
    shallow(<UploadStatus />);
  });

  it('Matches snapshot', async () => {
    const props = {
      uploadFile: file,
    };
    const wrapper = mount(<UploadStatus {...props} />);
    await flushPromises();
    wrapper.update();
    expect(toJson(wrapper.find('UploadStatus'))).toMatchSnapshot();
    wrapper.unmount();
  });
  it('Calls filereader', async () => {
    const props = {
      uploadFile: file,
    };
    const readAsDataURLSpy = jest.spyOn(FileReader.prototype, 'readAsDataURL');
    const wrapper = mount(<UploadStatus {...props} />);
    await flushPromises();
    wrapper.update();
    expect(readAsDataURLSpy).toBeCalledTimes(1);
  });
});
