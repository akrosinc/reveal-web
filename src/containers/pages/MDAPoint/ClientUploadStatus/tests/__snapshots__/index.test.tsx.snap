// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ClientUploadStatus Matches snapshot 1`] = `
<UploadStatus
  uploadFile={File {}}
>
  <p>
    loading...
  </p>
</UploadStatus>
`;
